src/NzbDrone.Core/Extras/Metadata/Consumers/Xbmc/XbmcMetadata.cs

```diff
@@ -101,120 +101,122 @@ public override MetadataFile FindMetadataFile(Movie movie, string path)
+                        video.Add(new XElement("codec", movieFile.MediaInfo.VideoCodec));
+                        video.Add(new XElement("framerate", movieFile.MediaInfo.VideoFps));
+                        video.Add(new XElement("height", movieFile.MediaInfo.Height));
+                        video.Add(new XElement("scantype", movieFile.MediaInfo.ScanType));
+                        video.Add(new XElement("width", movieFile.MediaInfo.Width));
+
+                        if (movieFile.MediaInfo.RunTime != null)
+                        {
+                            video.Add(new XElement("duration", movieFile.MediaInfo.RunTime.TotalMinutes));
+                            video.Add(new XElement("durationinseconds", movieFile.MediaInfo.RunTime.TotalSeconds));
+                        }
+
+                        streamDetails.Add(video);
+
+                        var audio = new XElement("audio");
+                        audio.Add(new XElement("bitrate", movieFile.MediaInfo.AudioBitrate));
+                        audio.Add(new XElement("channels", movieFile.MediaInfo.AudioChannels));
+                        audio.Add(new XElement("codec", GetAudioCodec(movieFile.MediaInfo.AudioFormat)));
+                        audio.Add(new XElement("language", movieFile.MediaInfo.AudioLanguages));
+                        streamDetails.Add(audio);
+
+                        if (movieFile.MediaInfo.Subtitles != null && movieFile.MediaInfo.Subtitles.Length > 0)
+                        {
+                            var subtitle = new XElement("subtitle");
+                            subtitle.Add(new XElement("language", movieFile.MediaInfo.Subtitles));
+                            streamDetails.Add(subtitle);
+                        }
+
+                        fileInfo.Add(streamDetails);
+                        details.Add(fileInfo);
                     }
 
-                    fileInfo.Add(streamDetails);
-                    details.Add(fileInfo);
-                }
+                    doc.Add(details);
+                    doc.Save(xw);
+
+                    xmlResult += doc.ToString();
+                    xmlResult += Environment.NewLine;
 
-                doc.Add(details);
-                doc.Save(xw);
+                }
+            }
+            if (Settings.MovieMetadataURL)
+            {
+                _logger.Debug("Generating Movie Metadata URL for: {0}", Path.Combine(movie.Path, movieFile.RelativePath));
 
-                xmlResult += doc.ToString();
+                xmlResult += "https://www.themoviedb.org/movie/" + movie.TmdbId;
                 xmlResult += Environment.NewLine;
 
+                xmlResult += "https://www.imdb.com/title/" + movie.ImdbId;
+                xmlResult += Environment.NewLine;
             }
 
             var metadataFileName = GetMovieMetadataFilename(movieFile.RelativePath);
 
-            if (Settings.UseMovieNfo)
-            {
-                metadataFileName = "movie.nfo";
```

Comment:
Why was this wrong? I didn't follow your description of how this lead to an error.

Reply:
IMHO metadataFileName should not be overridden here. It shall be properly defined in GetMovieMetadataFilename.

In PR #3096 when I a file rename was triggered the GetMovieMetadataFilename and the kodi metadata file was incorrectly renamed to the name of the movie.

Reply:
GetMovieMetadataFilename returns an absolute path. I had to build it for the movie.nfo case

Reply:
Because Path.ChangeExtension returns an absolute path. I did construct an absolute path for the movie.nfo as well to remain consistent with past behavior.

This code snipped behave consistently whether movieFilePath contains an absolute or a relative path. I do not remember exactly how & where but I noticed that depending on who called that method both type could be provided